### PR TITLE
Allow using sloths for bubblewrap.env values.

### PR DESCRIPTION
**Pull request by ccdunder**
Example usage:
```
        config = { sloth, ... }: {
          bubblewrap = {
            env = {
              "XDG_DATA_HOME" = sloth.xdgDataHome;
            };
          };
        };
```

### CODE REVIEW

File: modules/bubblewrap.nix [hunk 1]
```diff
@@ -14,30 +14,30 @@ let
     description = "pair of ${elemType.description}";
   };
 
-  bindType = with types; listOf (oneOf [
+  slothValue = types.oneOf [
```
Comment (max-privatevoid) [inline modules/bubblewrap.nix:17]: Renaming and changing this seems unnecessary. Why not use `sloth.type` in `env` below?

File: modules/bubblewrap.nix [hunk 2]
```diff
@@ -75,7 +75,7 @@ in {
 
     env = mkOption {
       description = "Environment variables to set.";
-      type = with types; attrsOf (nullOr str);
+      type = with types; attrsOf (oneOf [ (nullOr str) slothValue ]);
```
Comment (max-privatevoid) [inline modules/bubblewrap.nix:78]: I *think* this could just be `nullOr sloth.type`.

**Review by max-privatevoid — CHANGES_REQUESTED** [review]

File: modules/bubblewrap.nix (hunk 2 above)
Comment (ccdunder) [reply modules/bubblewrap.nix:78]: I tried but it doesn't work for all strings, only paths, per sloth.type.check.

File: modules/bubblewrap.nix (hunk 1 above)
Comment (ccdunder) [reply modules/bubblewrap.nix:17]: **tl;dr** Fixed.

My was trying to keep it DRY and support env var *name* sloths. But looking again, the env type def would need to be a lot more complex to do that and it doesn't seem worth it — dynamic env var names seem a very unusual usecase.

**Comment by ccdunder** [comment]
Thank you for the review @max-privatevoid! I think I addressed all the comments.

File: modules/bubblewrap.nix (hunk 2 above)
Comment (max-privatevoid) [reply modules/bubblewrap.nix]: Right, I guess sloth types should support any strings at this point, not just paths.

**Review by max-privatevoid — APPROVED** [review]

**Pull request merged**